File: entrega_5/Java/src/velocity/VelocityNormal.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package velocity;

import main_src.Particle;

/**
 *
 * @author Geraldo e Iago
 */
public class VelocityNormal extends AbstractVelocity{

    @Override
    public void calculate(Particle particle, int pos,
            double maxVel, int loopCount, int maxLoop)
    {
        double c1 = 2.05;
        double c2 = 2.05;
        
        double rand1 = this.getRandom();
        double rand2 = this.getRandom();
        
        double velocity = particle.getVelocity(pos);
        velocity += c1 * rand1 * (particle.getPBest(pos) - particle.getData(pos));
        velocity += c2 * rand2 * (particle.getGBest(pos) - particle.getData(pos));
        
        //clamping
        if(velocity > maxVel)
            velocity = maxVel;
        if(velocity < -maxVel)
            velocity = -maxVel;
        
        particle.setVelocity(pos, velocity);
    }
}

File: entrega_1/pso_simples/velocity/velocitynormal.cpp
#include "velocitynormal.h"

VelocityNormal::VelocityNormal()
{
    this->ran = new IagoRandom();
}

VelocityNormal::~VelocityNormal()
{
    delete this->ran;
}

void VelocityNormal::calculate(Particle *particle, int pos, double maxVel, int loopCount, int maxLoop)
{
    (void)loopCount;
    (void)maxLoop;

    double c1 = 2.05;
    double c2 = 2.05;

    double rand1 = ran->randNormal();
    double rand2 = ran->randNormal();

    double velocity = (particle->getVelocity(pos) +
                                 (c1 * rand1 * (particle->getPBest(pos) - particle->getData(pos))) +
                                 (c2 * rand2 * (particle->getGBest()->at(pos) - particle->getData(pos))));

    //clamping
    if(velocity > maxVel)
        velocity = maxVel;

    if(velocity < -maxVel)
        velocity = -maxVel;

    particle->setVelocity(pos, velocity);
}
